react-carousel-component/src/carousel2.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentIndex: 0,
+      previousIndex: props.images.length - 1,
+      nextIndex: 1,
```

Comment:
rather than store these, i recommend calculating them on demand

```js
getNextIndex() {

}

getPreviousIndex() {

}
```

it's really easy to overcomplicate code by storing anything that can be calculated on-the-fly. when you _do_ store something that can be calculated, it's called memoization (which is a form of caching) and it's a deliberate performance optimization.